sparkmagic/sparkmagic/livyclientlib/sparkcontroller.py

```diff
@@ -10,6 +10,10 @@
 
 
 class SparkController(object):
+    # this is to reuse the already created http clients
+    # since the reliablehttpclient uses requests session
+    _http_clients = {}
```

Comment:
Seems to better to have this be an attribute of `SparkController` instances, rather than class attribute? Can see class attribute leading to bugs, or at least confusion.

Reply:
Yes, it is better to have it as an instance attribute